felix/daemon/daemon.go

```diff
@@ -369,6 +369,11 @@ configRetry:
 		simulateDataRace()
 	}
 
+	// We may need to temporarily disable encrypted traffic to this node in order to connect to Typha
+	if configParams.WireguardEnabled && configParams.WireguardHostEncryptionEnabled {
+		bootstrapWireguard(configParams.WireguardInterfaceName, configParams.FelixHostname, v3Client)
```

Comment:
Do we need to clean up a key is wireguard _was_ enabled before but it no longer is?

Expecting some error handling here

Reply:
that's a good point, I should do the check regardless of if the current config says it's on/off!

Reply:
My expectation was that 
* You'd have wireguard routes in the dataplane on the local node.
* felix goes down
* Wreguard is turned off
* Other nodes disable wireguard, start sending unencrypted traffic
* Felix starts up, tries to connect to typha
* This node still has wireguard routes in place so we send encrypted traffic to a node that's not expecting it.

Reply:
Im beginning to think our only option for handling all these edge cases is to globally disable all WG on a node when felix goes up, or to pull back the hostnetworking feature altogether, though I hope Im wrong :grimacing: